src/core/annotation.js

```diff
@@ -2089,7 +2126,8 @@ class WidgetAnnotation extends Annotation {
       Array.isArray(value) &&
       Array.isArray(this.data.fieldValue) &&
       value.length === this.data.fieldValue.length &&
-      value.every((x, i) => x === this.data.fieldValue[i])
+      value.every((x, i) => x === this.data.fieldValue[i]) &&
```

Comment:
This is a pre-existing "issue", but there's a [helper function](https://github.com/mozilla/pdf.js/blob/783facb312c75941ba6bbfd5dffc52e85f46af5e/src/shared/util.js#L1017-L1027) we can use to simplify these two lines.
```javascript
isArrayEqual(value, this.data.fieldValue) &&
```